subsys/bluetooth/host/hci_raw.c

```diff
@@ -88,19 +130,162 @@ int bt_recv_prio(struct net_buf *buf)
 	return bt_recv(buf);
 }
 
+static int bt_h4_send(struct net_buf *buf)
+{
+	u8_t type;
+
+	type = net_buf_pull_u8(buf);
+
+	switch (type) {
+	case H4_CMD:
+		bt_buf_set_type(buf, BT_BUF_CMD);
+		break;
+	case H4_ACL:
+		bt_buf_set_type(buf, BT_BUF_ACL_OUT);
+		break;
+	default:
+		LOG_ERR("Unknown H4 type %u", type);
+		return -EINVAL;
+	}
+
+	LOG_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+
+	return 0;
+}
+
+static void bt_cmd_complete_ext(u16_t op, u8_t status)
+{
+	struct net_buf *buf;
+	struct bt_hci_evt_cc_status *cc;
+
+	if (status == BT_HCI_ERR_EXT_HANDLED) {
+		return;
+	}
+
+	buf = bt_hci_cmd_complete_create(op, sizeof(*cc));
+	cc = net_buf_add(buf, sizeof(*cc));
+	cc->status = status;
+
+	bt_recv(buf);
+}
+
+static u8_t bt_send_ext(struct net_buf *buf)
+{
+	struct bt_hci_cmd_hdr *hdr;
+	struct net_buf_simple_state state;
+	int i;
+	u16_t op;
+	u8_t status;
+
+	if (!cmd_ext) {
+		return status;
+	}
+
+	status = BT_HCI_ERR_SUCCESS;
+
+	net_buf_simple_save(&buf->b, &state);
+
+	if (buf->len < sizeof(*hdr)) {
+		BT_ERR("No HCI Command header");
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
+
+	hdr = net_buf_pull_mem(buf, sizeof(*hdr));
+	if (buf->len < hdr->param_len) {
+		BT_ERR("Invalid HCI CMD packet length");
+		return BT_HCI_ERR_INVALID_PARAM;
+	}
+
+	op = sys_le16_to_cpu(hdr->opcode);
+
+	for (i = 0; i < cmd_ext_size; i++) {
+		struct bt_raw_cmd_ext *cmd = &cmd_ext[i];
+
+		if (cmd->op == op) {
+			if (buf->len < cmd->min_len) {
+				status = BT_HCI_ERR_INVALID_PARAM;
+			} else {
+				status = cmd->func(buf);
+			}
+
+			break;
+		}
+	}
+
+	if (status) {
+		bt_cmd_complete_ext(op, status);
+		return status;
+	}
+
+	net_buf_simple_restore(&buf->b, &state);
+
+	return status;
+}
+
 int bt_send(struct net_buf *buf)
 {
 	BT_DBG("buf %p len %u", buf, buf->len);
 
+	if (IS_ENABLED(CONFIG_BT_HCI_RAW_H4) && raw_mode == BT_RAW_MODE_H4) {
+		int err = bt_h4_send(buf);
+
+		if (err) {
+			return err;
+		}
+	}
+
 	bt_monitor_send(bt_monitor_opcode(buf), buf->data, buf->len);
 
+	if (IS_ENABLED(CONFIG_BT_HCI_RAW_CMD_EXT) &&
+	    bt_buf_get_type(buf) == BT_BUF_CMD) {
+		u8_t status;
+
+		status = bt_send_ext(buf);
+		if (status) {
+			return 0;
+		}
```

Comment:
A code comment or debug log would be appropriate here. So `bt_send_ext()` is failing but you're treating it as success. Why?

Reply:
It should have been the status actually, this is now fixed.